pages/api/settings/awards.js

```diff
@@ -0,0 +1,50 @@
+import prisma from '@prisma/index'; // Prisma client
+import { getSession } from 'next-auth/client'; // Session handling
+
+// get awards by category IDs
+// return awards by category IDs
+export default async (req, res) => {
+  // Collect session from request
+  const session = await getSession({ req });
+  const { settingIDs } = req.body;
+
+  // If session exists (user authenticated)
+  if (session) {
+    // get awards
+    const awards = await getAwards(settingIDs);
+    console.log(awards);
+    // filtering in the API, TODO move to filtering via query
+    res.send(awards);
+  }
+
+  // Else, return 401 for all failures
+  res.status(401).end();
+};
+
+export const getAwards = async settingIDs => {
+  // Collect all awards from database
+  console.log(settingIDs);
+  const awards = await prisma.award.findMany({
+    include: {
+      awards_categories: {
+        include: {
+          settings: true,
+        },
+      },
+    },
+  });
+  return awards
```

Comment:
Maybe write a comment to explain what the map and filter is doing for future reference when we look back at this